Client/Frontend/Browser/BrowserViewController/OpenSearch/BrowserViewController+OpenSearch.swift

```diff
@@ -45,14 +45,17 @@ extension BrowserViewController {
         
         // Open Search guidlines requires Title to be same as Short Name but it is not enforced,
         // thus in case of yahoo.com the title is 'Yahoo Search' and Shortname is 'Yahoo'
-        // Instead we are checking referenceURL match to determine searchEngine is added or not
+        // We are checking referenceURL match to determine searchEngine is added or not
+        //In addition we are also checking if there is another engine with same name
         
         let searchEngineExists = profile.searchEngines.orderedEngines.contains(where: {
+            let nameExists = $0.shortName.lowercased() == referenceObject.title?.lowercased() ?? ""
```

Comment:
Do we have to unwrap `referenceObject` here? String to String? comparison should work, no?

Reply:
The Swift compiler does not implicitly unwrap any optionals as far as I know and It doesn't work I tried.